include/fmt/core.h

```diff
@@ -472,9 +472,11 @@ struct compile_string {};
 template <typename S>
 struct is_compile_string : std::is_base_of<compile_string, S> {};
 
-template <typename S>
-inline typename std::enable_if<is_compile_string<S>::value, string_view>::type
-  to_string_view(const S &s) { return {s.data(), s.size() - 1}; }
+template <typename S, typename Enable = typename std::enable_if<is_compile_string<S>::value>::type>
+inline auto to_string_view(const S &s) -> basic_string_view<typename S::char_type> { 
```

Comment:
Is there any reason to move `enable_if` from return type to template parameter?

Reply:
I am not sure if it is possible to use `S::char_type` inside `enable_if` since `S` may not satisfy `is_compile_string`, so I choose my familiar way to implement this.